local_pathfinding/obstacles.py

```diff
@@ -0,0 +1,340 @@
+                    to -708 to +708 degrees per minute
+        """
+        super().__init__(reference, sailbot_position, sailbot_speed)
+
+        self.id = ais_ship.id
+
+        # Position of the boat in XY
+        position = latlon_to_xy(
+            self.reference, LatLon(ais_ship.lat_lon.latitude, ais_ship.lat_lon.longitude)
+        )
+        self.position = position
+
+        # A Boat's collision zone is represented by a cone shaped polygon
+        self.collision_zone = self.create_collision_cone(
+            ais_ship.dimensions.width,
+            ais_ship.dimensions.length,
+            position,
+            ais_ship.sog.sog,
+            ais_ship.cog.cog,
+            ais_ship.rot.rot,
+            self.sailbot_position,
+            self.sailbot_speed,
+        )
+
+    def create_collision_cone(
+        self,
+        width: float,
+        length: float,
+        position: XY,
+        speed_over_ground: float,
+        course_over_ground: float,
+        rate_of_turn: float,
+        sailbot_position: XY,
+        sailbot_speed: float,
+    ) -> Polygon:
+        """
+        Creates a Shapely Polygon to represent the boat's collision_cone, which is sized,
+        orientated and positioned according to the boat's COG, SOG, and position.
+
+        The polygon is oversized according to the collision zone safety buffer, for
+        added assurrance that the boat will be entirely contained by the polygon.
+
+        Args:
+            width (float): width of the boat in meters
+            length (float): length of the boat in meters
+            position (XY): x,y coordinates of the boat in km
+            speed_over_ground (float): speed of the boat in knots, over ground
+            course_over_ground (float): COG of the boat in degrees clockwise from true north
+            rate_of_turn (float): ROT of the boat in AISROT scale -126 to +126 corresponding to
+                -708 to +708 degrees per minute
+            Sailbot_position (XY): x,y coordinates of the Sailbot in km
+
+        Notes:
+            ROT is not incorporated yet, but may be in the future.
+        """
+
+        # coordinates of the center of the boat
+        x, y = position[0], position[1]
+
+        speed_over_ground_kmph = knots_to_kilometers_per_hour(speed_over_ground)
+
+        width = meters_to_km(width)
+        length = meters_to_km(length)
+
+        # Calculate distance the boat will travel before soonest possible collision with Sailbot
+        projected_distance = self.calculate_projected_distance(
+            position,
+            course_over_ground,
+            speed_over_ground_kmph,
+            sailbot_position,
+            sailbot_speed,
+        )
+
+        # This factor can be adjusted to change the scope/width of the collision cone
+        # TODO This feels too arbitrary, maybe will incorporate ROT at a later time
+        COLLISION_CONE_STRETCH_FACTOR = projected_distance * 1.5
```

Comment:
All caps reserved for constants defined at the top of the file. Can make `1.5` a constant